sim/core/aura.go

```diff
@@ -35,41 +35,83 @@ type Aura struct {
 	OnExpire       func(sim *Simulation)
 }
 
-func NewAuraTracker() *AuraTracker {
-	return &AuraTracker{
-		ActiveAuraIDs: make([]int32, 0, 5),
+// This needs to be a function that returns an Aura rather than an Aura, so captured
+// local variables can be reset on Sim reset.
+type PermanentAura func(*Simulation) Aura
+
+// auraTracker is a centralized implementation of CD and Aura tracking.
+//  This is currently used by Player and Raid (for global debuffs)
+type auraTracker struct {
```

Comment:
it seems weird to make this private. Since many of its functions are public and they will be exposed on public structs, why would we make it a private type that people are still expected to use?